[GCC 8] error: catching polymorphic type by value

I ran into this problem compiling Rust, but reproduced it on binaryen master (c81857a3a7708738b20bb28a320fc971e74626a7).

Compiling with GCC 8 on Fedora rawhide gives the following errors:

```
[  5%] Building CXX object src/wasm/CMakeFiles/wasm.dir/wasm-s-parser.cpp.o
/builddir/binaryen/src/wasm/wasm-s-parser.cpp: In member function ‘wasm::Name wasm::SExpressionWasmBuilder::getLabel(wasm::Element&)’:
/builddir/binaryen/src/wasm/wasm-s-parser.cpp:1411:19: error: catching polymorphic type ‘class std::invalid_argument’ by value [-Werror=catch-value=]
     } catch (std::invalid_argument) {
                   ^~~~~~~~~~~~~~~~
/builddir/binaryen/src/wasm/wasm-s-parser.cpp:1413:19: error: catching polymorphic type ‘class std::out_of_range’ by value [-Werror=catch-value=]
     } catch (std::out_of_range) {
                   ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[3]: *** [src/wasm/CMakeFiles/wasm.dir/build.make:159: src/wasm/CMakeFiles/wasm.dir/wasm-s-parser.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:1012: src/wasm/CMakeFiles/wasm.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:215: CMakeFiles/binaryen.dir/rule] Error 2
gmake: *** [Makefile:203: binaryen] Error 2
```

However, compiling without `-Werror` is successful. (cc #1387)

--- a/src/tools/asm2wasm.cpp
+++ b/src/tools/asm2wasm.cpp
@@ -87,7 +87,7 @@ int main(int argc, const char *argv[]) {
            [&trapMode](Options *o, const std::string &argument) {
              try {
                trapMode = trapModeFromString(argument);
-             } catch (std::invalid_argument e) {
+             } catch (std::invalid_argument& e) {
                std::cerr << "Error: " << e.what() << "\n";
                exit(EXIT_FAILURE);
              }

--- a/src/tools/s2wasm.cpp
+++ b/src/tools/s2wasm.cpp
@@ -92,7 +92,7 @@ int main(int argc, const char *argv[]) {
            [&trapMode](Options *o, const std::string &argument) {
              try {
                trapMode = trapModeFromString(argument);
-             } catch (std::invalid_argument e) {
+             } catch (std::invalid_argument& e) {
                std::cerr << "Error: " << e.what() << "\n";
                exit(EXIT_FAILURE);
              }

--- a/src/wasm/wasm-s-parser.cpp
+++ b/src/wasm/wasm-s-parser.cpp
@@ -1408,9 +1408,9 @@ Name SExpressionWasmBuilder::getLabel(Element& s) {
     uint64_t offset;
     try {
       offset = std::stoll(s.c_str(), nullptr, 0);
-    } catch (std::invalid_argument) {
+    } catch (std::invalid_argument&) {
       throw ParseException("invalid break offset");
-    } catch (std::out_of_range) {
+    } catch (std::out_of_range&) {
       throw ParseException("out of range break offset");
     }
     if (offset > nameMapper.labelStack.size()) throw ParseException("invalid label", s.line, s.col);

